lib/coverage.js

```diff
@@ -49,10 +49,29 @@ exports.instrument = function (options) {
 
 internals.pattern = function (options) {
 
-    const base = internals.escape(options.coveragePath || '');
-    let excludes = options.coverageExclude ? [].concat(options.coverageExclude).map(internals.escape) : '';
-    excludes = excludes ? excludes.map((exclude) => `${exclude}\\/`).join('|') : excludes;
+    const coveragePath = options.coveragePath || '';
+    const base = internals.escape(coveragePath);
+    let excludes = options.coverageExclude ? [].concat(options.coverageExclude).map((path) => {
+
+        let isFile = false;
+        try {
+            const pathStat = Fs.statSync(Path.join(coveragePath, path));
+            isFile = pathStat.isFile();
+        }
+        // ok to swallow the error as these can be sub folders, or not even exist (which is ok)
+        catch (ex) {}
+
+        return {
+            isFile,
+            escaped: internals.escape(path)
+        };
+    }) : '';
```

Comment:
It might just be me, but the large `map()` function combined with the ternary makes this a bit harder to read.